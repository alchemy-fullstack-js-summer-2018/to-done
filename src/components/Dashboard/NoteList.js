import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class NoteList extends Component {

  static propTypes = {
    notes: PropTypes.array
  };

  render() {
    const { notes } = this.props;
    if(!notes) return null;

    return (
      <ul>
        {notes.map(note => (
          <Note
            key={note.key}
            note={note}
          />
        ))}
      </ul>
    );
  }
}

export default NoteList;