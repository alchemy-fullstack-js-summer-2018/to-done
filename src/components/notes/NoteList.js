import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class NoteList extends Component {
  static propTypes = {
    noteList: PropTypes.array,
    onUpdate: PropTypes.func.isRequired,
    onRemove: PropTypes.func.isRequired
  };

  render() {
    const { noteList, onUpdate, onRemove } = this.props;

    return (
      <ul>
        {noteList.map(note => (
          <Note
            key={note.key}
            note={note}
            onUpdate={onUpdate}
            onRemove={onRemove}
          />
        ))}
      </ul>
    );
  }
}
 
export default NoteList;