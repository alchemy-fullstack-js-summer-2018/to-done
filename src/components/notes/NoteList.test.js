import React from 'react';
import { shallow } from 'enzyme';
import NoteList from './NoteList';
import toJSON from 'enzyme-to-json';

describe('Notes List', () => {
  it('renders add if no note prop', () => {
    const noteList = [
      { title: 'Get milk', content: 'Must get milk', key: 'abc' },
      { title: 'Get eggs', content: 'Must get eggs', key: 'def' },
      { title: 'Get coffee', content: 'Must get coffee', key: 'ghi' }
    ];

    const wrapper = shallow(<NoteList
      noteList={noteList}
      onUpdate={jest.fn()}
      onRemove={jest.fn()}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});