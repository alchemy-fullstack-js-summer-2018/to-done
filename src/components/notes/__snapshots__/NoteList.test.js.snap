// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes List renders add if no note prop 1`] = `
<ul>
  <Note
    key="abc"
    note={
      Object {
        "content": "Must get milk",
        "key": "abc",
        "title": "Get milk",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="def"
    note={
      Object {
        "content": "Must get eggs",
        "key": "def",
        "title": "Get eggs",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="ghi"
    note={
      Object {
        "content": "Must get coffee",
        "key": "ghi",
        "title": "Get coffee",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
</ul>
`;
